src/Idris/Parser/Data.hs

```diff
@@ -30,7 +31,8 @@ Record ::=
     DocComment Accessibility? 'record' FnName TypeSig 'where' OpenBlock Constructor KeepTerminator CloseBlock;
 -}
 record :: SyntaxInfo -> IdrisParser PDecl
-record syn = do (doc, paramDocs, acc, opts) <- P.try (do
+record syn = (<?> "record type declaration") . appExtent $ do
```

Comment:
This looks a bit funky, is there any reason why this is not added afterwards?

Reply:
Not actually a good one, in the end.  Will fix.